### PR TITLE
When editing users, full name now follows first and last name

### PR DESCRIPTION
**Pull request by boutell**
* Full name now follows first and last name input when editing a user
* `following` may now be a single field name or an array of field names
* `followingValue` prop of input components is now `followingValues`, an object with sub-properties for all the fields your field is following
* AposInputString will join the followed values with spaces and use that, provided the old value was already a match for the old followed value (in sync, not deliberately different)
* AposInputSlug now supports multiple source fields in the same way, because why not
* fixed the slug prefix for users and groups
* removed the published column and filter for users and groups

### CODE REVIEW

File: modules/@apostrophecms/modal/ui/apos/mixins/AposEditorMixin.js
Comment (abea) [inline modules/@apostrophecms/modal/ui/apos/mixins/AposEditorMixin.js:34]: We do have some editors that don't have these split groups (e.g., media editor) that do the following dance.

Comment (abea) [reply modules/@apostrophecms/modal/ui/apos/mixins/AposEditorMixin.js:34]: If we have an `else` in here using `self.doc` that should do it. Also another argument for us to undo this splitDoc business.

File: modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue
Comment (abea) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue:91]: Should we bail of the values are objects or arrays?

**Review by abea — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue
Comment (boutell) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue:91]: The only values we will ever receive are those we are `following`. This input field type should be documented as having support for `following` fields that convert themselves sensibly to a string. I don't think we need to do more than that.

Comment (boutell) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue:91]: (Documentation being a task we all know is forthcoming.)

**Comment by abea** [comment]
@boutell Did you see the other comment about the media editor not splitting the doc?

**Comment by boutell** [comment]
@abea I addressed this with some changes to the mixin and the use of the method override pattern as a way to migrate any weirdness to things that are actually weird, i.e. AposDocEditor.

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue
Comment (boutell) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:106]: Now docFields, which is configured for us by the mixin.

Comment (boutell) [reply modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:106]: I prefer docFields because (1) it's analogous to the split field properties, and (2) doc should be a doc, not a thing with a data property and an error property, plus it's not uncommon for the actual existing doc to be a prop in other potential users of the mixin.

File: modules/@apostrophecms/schema/ui/apos/components/AposInputString.vue
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputString.vue:90]: Trim solved a problem when first or last name is blank.

File: modules/@apostrophecms/widget-type/ui/apos/components/AposWidgetEditor.vue
Comment (abea) [inline modules/@apostrophecms/widget-type/ui/apos/components/AposWidgetEditor.vue:18]: This `ref` isn't used, right? Can we nix that before merge?

**Comment by boutell** [comment]
Marked as a draft because I merged the as yet unapproved esc key branch into it, so I'm not blocked on that later when the esc key branch is approved.

File: modules/@apostrophecms/widget-type/ui/apos/components/AposWidgetEditor.vue
Comment (abea) [reply modules/@apostrophecms/widget-type/ui/apos/components/AposWidgetEditor.vue:19]: It's used in the mixin

Comment (boutell) [reply modules/@apostrophecms/widget-type/ui/apos/components/AposWidgetEditor.vue:19]: Yep

**Review by abea — APPROVED** [review]

**Pull request merged**
